src/builder.rs

```diff
@@ -206,14 +217,30 @@ pub enum InitError<DI, P> {
     /// This error is returned when the configuration passed to the builder is
     /// invalid. For example, when the combination of bit depth and interface
     /// kind isn't supported by the selected model.
-    InvalidConfiguration,
+    InvalidConfiguration(ConfigurationError),
+}
+
+/// Specifics of [InitError::InvalidConfiguration] if configuration was found invalid
+#[non_exhaustive]
+#[derive(Debug)]
+pub enum ConfigurationError {
+    /// Display Model and Interface kind are mismatching
+    ModelInterfaceMismatch,
```

Comment:
I think it's important to mention that changing the color format might fix this issue:
```suggestion
    /// Unsupported interface kind.
    /// 
    /// The chosen interface isn't supported by the selected model. Note that
    /// some controller models don't support all combinations of physical
    /// interface and color formats. To resolve this, try to use another color
    /// format if available (e.g.
    /// [`ILI9486Rgb666`](crate::models::ILI9486Rgb666) instead of
    /// [`ILI9486Rgb565`](crate::models::ILI9486Rgb565) if you use a SPI
    /// connection).
    UnsupportedInterface,
```